internal/web/log.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2021 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package web
+
+import (
+	"io"
+	"net/http"
+	"regexp"
+	"strconv"
+	"sync"
+
+	"github.com/labstack/echo/v4"
+	"github.com/tsuru/rpaas-operator/internal/pkg/rpaas"
+)
+
+type flushWriter struct {
+	sync.Mutex
+
+	w io.Writer
+}
+
+func (fw *flushWriter) Write(p []byte) (int, error) {
+	fw.Lock()
+	defer fw.Unlock()
+
+	n, err := fw.w.Write(p)
+	if err != nil {
+		return n, err
+	}
+
+	if f, ok := fw.w.(http.Flusher); ok && fw.w != nil {
+		f.Flush()
+	}
+	return n, nil
+}
+
+func parseQueries(pod, container string) (map[string]*regexp.Regexp, error) {
+	var p *regexp.Regexp
+	var c *regexp.Regexp
+	var err error
+
+	everything, err := regexp.Compile(`.*`)
+	if err != nil {
+		return nil, err
+	}
+
+	if pod != "" {
+		p, err = regexp.Compile(pod)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		p = everything
+	}
+
+	if container != "" {
+		c, err = regexp.Compile(container)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		c = everything
+	}
+
+	return map[string]*regexp.Regexp{
+		"pod":       p,
+		"container": c,
+	}, nil
+}
+
+func extractLogArgs(c echo.Context) (rpaas.LogArgs, error) {
+	params := c.Request().URL.Query()
+	var err error
+
+	var pLines *int64
+	lines, _ := strconv.ParseInt(params.Get("lines"), 10, 64)
+	if lines > 0 {
+		pLines = &lines
+	}
+
+	tSince := int64(172800) // last 48 hours by default -> 48 * 60(minutes) * 60(seconds)
```

Comment:
Last 48h might return too much logs, let's try 24h.

```suggestion
	tSince := int64(24 * 60 * 60) // last 24 hours by default
```